internal/pkg/template/templates/addons/s3/env/cf.yml

```diff
@@ -0,0 +1,54 @@
+Parameters:
+  App:
+    Type: String
+    Description: Your application's name.
+  Env:
+    Type: String
+    Description: The name of the environment being deployed.
+
+Resources:
+  {{logicalIDSafe .Name}}Bucket:
+    Metadata:
+      'aws:copilot:description': 'An Amazon S3 bucket to store and retrieve objects for {{.Name}}'
+    Type: AWS::S3::Bucket
+    Properties:
+      AccessControl: Private
+      BucketEncryption:
+        ServerSideEncryptionConfiguration:
+        - ServerSideEncryptionByDefault:
+            SSEAlgorithm: AES256
+      PublicAccessBlockConfiguration:
+        BlockPublicAcls: true
+        BlockPublicPolicy: true
+
+  {{logicalIDSafe .Name}}BucketPolicy:
+    Metadata:
+      'aws:copilot:description': 'A bucket policy to deny unencrypted access to the bucket and its contents'
+    Type: AWS::S3::BucketPolicy
+    DeletionPolicy: Retain
```

Comment:
why do we retain the bucket policy?

Reply:
good question!!! It was initially added by [this commit](https://github.com/aws/copilot-cli/pull/1108/files#diff-f7921c17445c4b5d5487a2de5974851ba2a710e56f491f66d030c01ffc86877fR24) for workload addons.

My guess is that since this bucket policy has a dependency on the bucket, it will gets deleted before the bucket. This creates a window during which the bucket exists while the SSL connection is not enforced (because the policy has been deleted). To prevent this from happening, we add the retain policy.

I can double check with @bvtujo ! But this is my guess!

Reply:
Gotcha. It's ok they can update the template themselves and we just promote the best practice.

Reply:
Wouldn't `DependsOn` instead handle this problem rather than `DeletionPolicy: Retain`? 🥺

Reply:
Do you mean like this?
```yaml
{{logicalIDSafe .Name}}Bucket:
    Metadata:
      'aws:copilot:description': 'An Amazon S3 bucket, {{.Name}}, for storing and retrieving objects'
    Type: AWS::S3::Bucket
    DependsOn: {{logicalIDSafe .Name}}BucketPolicy
```

That would introduce circular dependency! BucketPolicy has `!Sub ${ {{logicalIDSafe .Name}}Bucket.Arn}/*`

Reply:
nono i meant the other way around:
```
 {{logicalIDSafe .Name}}BucketPolicy:
    Metadata:
      'aws:copilot:description': 'A bucket policy to deny unencrypted access to the bucket and its contents'
    Type: AWS::S3::BucketPolicy
    DependsOn: {{logicalIDSafe .Name}}Bucket
```

Wouldn't that solve the concern in the above comment?